matchzoo/datapack.py

```diff
@@ -0,0 +1,114 @@
+"""Matchzoo DataPack, pair-wise tuple (feature) and context as input."""
+
+import typing
+from pathlib import Path
+
+import dill
+import pandas as pd
+
+
+class DataPack(pd.DataFrame):
+    """
+    Matchzoo DataPack data structure, store dataframe and context.
+
+    Example:
+        >>> # features, context generate by processors.
+        >>> features = [([1,3], [2,3]), ([3,0], [1,6])]
+        >>> context = {'vocab_size': 2000}
+        >>> dp = DataPack(data=features,
+        ...               context=context)
+        >>> dp.context
+        {'vocab_size': 2000}
+        >>> # sample without replacement for generation.
+        >>> type(dp.sample(1))
+        <class 'matchzoo.datapack.DataPack'>
+        >>> dp.size
+        2
+        >>> features, context = dp.unpack()
+    """
+
+    _metadata = ['context']
+
+    DATA_FILENAME = 'data.pkl'
```

Comment:
Just checked my `base_model.py` and it has the same problem: when you save with `dill`, the filename should be something like `data.dill`.

Reply:
got it